mapper/src/main/java/com/ibm/mapper/model/algorithms/AES.java

```diff
@@ -44,24 +44,23 @@ public final class AES extends Algorithm implements BlockCipher, AuthenticatedEn
     }
 
     public AES(@NotNull DetectionLocation detectionLocation) {
-        super(NAME, BlockCipher.class, detectionLocation);
-        this.put(new BlockSize(128, detectionLocation));
+        this(BlockCipher.class, detectionLocation);
     }
 
     public AES(int keyLength, @NotNull DetectionLocation detectionLocation) {
-        super(NAME, BlockCipher.class, detectionLocation);
+        this(detectionLocation);
         this.put(new KeyLength(keyLength, detectionLocation));
         this.put(new BlockSize(128, detectionLocation));
     }
 
     public AES(@Nonnull Mode mode, @NotNull DetectionLocation detectionLocation) {
-        super(NAME, BlockCipher.class, detectionLocation);
+        this(detectionLocation);
         this.put(new BlockSize(128, detectionLocation));
         this.put(mode);
     }
 
     public AES(int keyLength, @Nonnull Mode mode, @NotNull DetectionLocation detectionLocation) {
-        super(NAME, BlockCipher.class, detectionLocation);
+        this(detectionLocation);
         this.put(new KeyLength(keyLength, detectionLocation));
         this.put(new BlockSize(128, detectionLocation));
         this.put(mode);
```

Comment:
I think the `this.put(new BlockSize(128, detectionLocation));` shouldn't be there, as calling `this(detectionLocation)` will itself call `this(BlockCipher.class, detectionLocation)`, which is already adding the block size